src/features/ranking/service/rankingOptions.ts

```diff
@@ -0,0 +1,17 @@
+interface RankingOption {
+  icon: string;
+  dataField: 'level' | 'point';
+}
+
+const rankingOptions: Record<string, RankingOption> = {
+  '포인트 보유순': {
+    icon: '포인트.svg',
+    dataField: 'point', // 포인트 기준
+  },
+  '레벨순': {
+    icon: '과일바구니.svg',
+    dataField: 'level', // 레벨 기준
+  },
+};
```

Comment:
```suggestion
const rankingOptions = {
  '포인트 보유순': {
    icon: '포인트.svg',
    dataField: 'point', // 포인트 기준
  },
  '레벨순': {
    icon: '과일바구니.svg',
    dataField: 'level', // 레벨 기준
  },
} as const;
```
@bluetree7878  이런 변하지 않을 객체들은 `as const`를 통해 명확하게 관리하는게 좋습니다.
타입스크립트를 다룰 때 가능하면 좁은 타입으로 관리를 해보세요.

Reply:
리뷰 감사합니다! 전에 리뷰 달아주셨을 때 똑같이 `as const`에 관한 리뷰를 본 적이 있었어서 똑같이 적용해보았으나
```
'string' 형식의 식을 '{ readonly '\uD3EC\uC778\uD2B8 \uBCF4\uC720\uC21C': { readonly icon: "포인트.svg"; readonly dataField: "point"; }; readonly 레벨순: { readonly icon: "과일바구니.svg"; readonly dataField: "level"; }; }' 인덱스 형식에 사용할 수 없으므로 요소에 암시적으로 'any' 형식이 있습니다.
  '{ readonly '\uD3EC\uC778\uD2B8 \uBCF4\uC720\uC21C': { readonly icon: "포인트.svg"; readonly dataField: "point"; }; readonly 레벨순: { readonly icon: "과일바구니.svg"; readonly dataField: "level"; }; }' 형식에서 'string' 형식의 매개 변수가 포함된 인덱스 시그니처를 찾을 수 없습니다.
```
위와같이 해당 오류가 나길래 기존에 썼던 방식으로 돌아갔던 건데 @rhehfl 님께서 잘못된 부분을 잘 짚어주셨습니다.

```tsx
interface RankingContainerProps {
  myRank: Rank;
  selectedOption: string;
  onOptionChange: (option: string) => void;
  users: User[];
}
```
타입을 이렇게 지정해뒀던 게 문제가 되었네요. 

```tsx
selectedOption: '포인트 보유순' | '레벨순';
```
따라서 문제 짚어주신 걸 이렇게 수정하니까 해당 오류는 사라지고 `as const`를 써도 문제가 없어졌습니다... 
@rhehfl @ssi02014 두 분께 정말 감사드립니다😂😂

Reply:
@bluetree7878 네! 좁은 타입으로 다룰 수 있는데 타입 에러가 있는 거라면 보통 실수했다거나/개선 할 수 있는 부분이 있을 것 같습니다 ㅎㅎ 
지금도 좋지만 더 좋은 방법이라면, 하드 코딩하는게 아닌 아래와 같은 방식이 될 것 같습니다. 타입도 중복/하드 코딩은 없애는게 좋습니다 🤗

`selectedOption: keyof rankingOptions`

추가로 네이밍도 `RANKING_OPTIONS`도 나쁘지 않을 것 같습니다. ㅎㅎ